membership-attribute-service/app/components/TouchpointComponents.scala

```diff
@@ -121,18 +122,27 @@ class TouchpointComponents(
     zuoraRestServiceOverride.getOrElse(simpleClientZuoraRestService)
   }
 
-  lazy val catalogRestClient = rest.SimpleClient[Future](backendConfig.zuoraRest, RequestRunners.configurableFutureRunner(60.seconds))
-  lazy val futureCatalog: Future[Catalog] = catalogServiceOverride.getOrElse(
-    CatalogService
-      .read(FetchCatalog.fromZuoraApi(catalogRestClient)(implicitly, LogPrefix.noLogPrefix), productIds)
+  private lazy val futureCatalogNoPrefix: Future[Catalog] = {
+    logger.infoNoPrefix(s"Loading catalog in $stage")
+    val catalogRestClient = rest.SimpleClient[Future](backendConfig.zuoraRest, RequestRunners.configurableFutureRunner(60.seconds))
+    catalogServiceOverride.getOrElse(
+      CatalogService
+        .read(FetchCatalog.fromZuoraApi(catalogRestClient)(implicitly, LogPrefix.noLogPrefix), productIds)
+        .flatMap {
+          case -\/(error) => Future.failed(new RuntimeException(error))
+          case \/-(result) => Future.successful(result)
+        },
```

Comment:
this pushes "left"s into a failed Future, as we don't really need to keep it as a `Future[Either....]]` at this point